### PR TITLE
clean up demo. add jump.js. make configurable

### PR DESCRIPTION
**Pull request by yutakahoulette-zz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,23 @@
 # anchor-scroll-hash
 
-This is a plain-JS library for enhancing anchor hash link behavior:
+### [Demo](http://jayrbolton.github.io/anchor-hash-scroll)
```
Comment (yutakahoulette-zz) [inline README.md:3]: add link to demo page

File: README.md [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 
-* Smooth scrolls to the linked element
-* Tracks user scrolling and sets `data-active` attributes to be true for the link and section elements within the current scroll
-* Tracks user scrolling and changes current hash in url based on scroll position
+### Features:
```
Comment (yutakahoulette-zz) [inline README.md:5]: update readme

File: docs/build.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
+(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
```
Comment (yutakahoulette-zz) [inline docs/build.js:1]: built js for demo page

File: docs/index.css [hunk 4]
```diff
@@ -0,0 +1,72 @@
+* { box-sizing: border-box; }
```
Comment (yutakahoulette-zz) [inline docs/index.css:1]: styles for demo page

File: docs/index.html [hunk 5]
```diff
@@ -1,35 +1,21 @@
...
-  <script src='file:///home/big/j/Sync/code/anchor-hash-scroll/index.js'></script>
-</body>
+  <head>
+    <meta charset="utf-8">
+    <title>Anchor hash scroll</title>
+    <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css">
+    <link rel="stylesheet" type="text/css" href="./index.css">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+  </head>
+  <body>
```
Comment (yutakahoulette-zz) [inline docs/index.html:9]: generated markup with js

File: docs/index.html [hunk 6]
```diff
@@ -1,35 +1,21 @@
...
+    <div style='padding: 4rem 1rem'>
+      <h1>Anchor hash scroll demo</h1>
+      <p>Scroll, or click the links at the top.</p>
+      <div style='padding-top: 2rem'>
+        <a class="github-button" href="https://github.com/jayrbolton/anchor-hash-scroll" data-icon="octicon-star" data-style="mega" aria-label="Star jayrbolton/anchor-hash-scroll on GitHub">Star</a>
```
Comment (yutakahoulette-zz) [inline docs/index.html:14]: added github star button 😛

File: index.js [hunk 7]
```diff
@@ -18,66 +11,58 @@ function initialize() {
...
+    if(elems[i].section.offsetTop <= scrollPos) found = i
     else passed = true
   }
-
   var elem = elems[found]
-  if(elem && elem.id !== currentHash) activateElem(elems, found)
+  if(elem && elem.id !== currentHash) activateElem(elems, found, currentHash)
 }
+
+module.exports = function (config) {
```
Comment (yutakahoulette-zz) [inline index.js:62]: exports a function that takes a config object (for jump.js and offset)

File: package.json [hunk 8]
```diff
@@ -1,12 +1,13 @@
 {
   "name": "anchor-hash-scroll",
-  "version": "0.0.2",
+  "version": "0.0.3",
```
Comment (yutakahoulette-zz) [inline package.json:3]: bump version

File: docs/build.js [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+function init(config) {
+  // Find all anchor links that have a hash href
+  var currentHash = null
+  var sel = 'a[href^="#"]'
+  var anchors = document.querySelectorAll(sel)
+  var elems = []
+  for(var i = 0 ; i < anchors.length ; ++i) {
+    var anchor = anchors[i]
+    var id = anchor.getAttribute('href')
+    var section = document.querySelector('[id="' + id.replace('#', '') + '"]')
```
Comment (jayrbolton) [inline docs/build.js]: can you just do `var section = document.querySelector(id)` ?

File: index.js [hunk 10]
```diff
@@ -18,66 +11,58 @@ function initialize() {
     var id = anchor.getAttribute('href')
     var section = document.querySelector('[id="' + id.replace('#', '') + '"]')
```
Comment (jayrbolton) [inline index.js]: can you just do `var section = document.querySelector(id)` ?

**Comment by jayrbolton** [comment]
Looking good

**Pull request merged**